From 341e6a1cbbc8a06702df15c147cbe1264418bf3b Mon Sep 17 00:00:00 2001
From: Biwen Li <biwen.li@nxp.com>
Date: Thu, 5 Dec 2019 17:21:06 +0800
Subject: [PATCH] soc: fsl: rcpm: remove build warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This removes build warning,
drivers/soc/fsl/rcpm.c: In function ‘rcpm_pm_prepare’:
drivers/soc/fsl/rcpm.c:126:37: warning: left shift count >= width of type [-Wshift-count-overflow]
          (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
                                     ^~
drivers/soc/fsl/rcpm.c:131:38: warning: left shift count >= width of type [-Wshift-count-overflow]
           (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |

Reviewed-by: Ran Wang <ran.wang_1@nxp.com>
Signed-off-by: Biwen Li <biwen.li@nxp.com>
---
 drivers/soc/fsl/rcpm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/soc/fsl/rcpm.c
+++ b/drivers/soc/fsl/rcpm.c
@@ -123,12 +123,12 @@ static int rcpm_pm_prepare(struct device
 				}
 				/* Read value from register SCFG_SPARECR8 */
 				regmap_read(scfg_addr_regmap,
-					    (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
+					    ((((u64)reg_offset[1] << (sizeof(u32) * 8) |
 					    reg_offset[2])) & 0xffffffff),
 					    &reg_value);
 				/* Write value to register SCFG_SPARECR8 */
 				regmap_write(scfg_addr_regmap,
-					     (u32)(((u64)(reg_offset[1] << (sizeof(u32) * 8) |
+					     ((((u64)reg_offset[1] << (sizeof(u32) * 8) |
 					     reg_offset[2])) & 0xffffffff),
 					     tmp | reg_value);
 			}
